docs/meshstack.customer-group-sync.md

```diff
@@ -4,3 +4,206 @@ title: meshCustomer Group Synchronisation
+
+## Configuration Reference
+
+This section describes how to configure the LDAP group synchronization in meshStack.
+For easier reference the following sections break down the configuration model in multiple parts. The union of these
+defines the full configuration model.
+
+The basic parameters required to configure the connection to LDAP are as follows.
+
+<!--snippet:mesh.identityconnector.ldap.core#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let ConnectionConfiguration =
+    {-
+      url: The LDAP connection URL which would be of the form ldap://example.com:389
+      base:
+          The base from which searches should be performed, which would be of the form "dc=meshcloud,dc=io"
+      username: The username of the user who can connect to the LDAP source to perform searches
+      password: The password of the user above
+    -}
+      { url : Text, base : Text, username : Text, password : Secret }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The configuration related to the collection of data via LDAP is as follows.
+
+<!--snippet:mesh.identityconnector.ldap.collector#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+{-
+  pageSize: The size of a single page that will be returned as a result of an LDAP search
+  group/person:
+    a configuration of type LDAPFilter which will contain the information needed to search the LDAP source
+    for users and groups.
+-}
+  let CollectorConfiguration =
+        { pageSize : Natural, group : LdapFilter, person : LdapFilter }
+
+  in  CollectorConfiguration
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The LdapFilter mentioned above has the following form.
+
+<!--snippet:mesh.identityconnector.ldap.filter#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+{-
+    attributes:
+        A comma separated list of attributes that should be returned per LDAP entry.
+        For example, "cn, uniqueMember, description"
+    base: The base from which the query should be performed. For example, "ou=groups"
+    filter: A filter that follows the LDAP search filter format. For example, "'(cn=mc*)'"
+-}
+  let LdapFilter = { attributes : Text, base : Text, filter : Text }
+
+  in  LdapFilter
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+
+The following configuration describes how the collected data should be transformed into meshObjects.
+<!--snippet:mesh.identityconnector.ldap.transformation#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let TransformationConfiguration =
+    {-
+    mesh-object:
+      The type of meshObject this configuration is about. Can be one of
+      MeshUser, MeshCustomerUserGroup or MeshCustomerGroupBinding
```

Comment:
```suggestion
      meshUser, meshCustomerUserGroup or meshCustomerGroupBinding
```

Reply:
hm no the names given here I think are enum values which have this convention of being uppercase at the beginning. So that are indeed the legal values. Maybe put this into code tags so its clear we deal here not with our business names but with actual values.
```suggestion
      `MeshUser`, `MeshCustomerUserGroup` or `MeshCustomerGroupBinding`
```
oh I see its insie a dhall code block... well then good luck. :D Seems we need to keep it like this. Maybe quote it with " at least.

Reply:
Will add quotes.